tests/integration/scrapers/scrapers-all-test.js

```diff
@@ -1,94 +1,172 @@
+const path = require('path');
+const { EventEmitter } = require('events');
+
+const fs = imports(join(process.cwd(), 'src', 'shared', 'lib', 'fs.js'));
+const runScraper = imports(join(process.cwd(), 'src', 'events', 'crawler', 'scrape-data', 'run-scraper.js'));
+const get = imports(join(process.cwd(), 'src', 'shared', 'lib', 'fetch', 'get.js'));
+
+// This suite automatically tests a scraper's results against its test
+// cases. To add test coverage for a scraper, see
+// docs/sources.md#testing-sources
+
+// The tests monkeypatch get.get, so make sure that's restored at the end!
+const oldGetGet = get.get;
+
+// Utility functions
+
+/** Splits folder path into the scraper name and test date hash.
+ * e.g. 'X/Y/2020-03-04' => { scraperName: 'X/Y', date: '2020-03-04' }
  */
-// Extract date from filename, e.g. `expected-2020-03-16.json` -> `2020-03-16`
-const datedResultsRegex = /expected.(\d{4}-\d{2}-\d{2}).json/i;
-const getDateFromPath = path => datedResultsRegex.exec(path, '$1')[1];
-
-// e.g. `/coronadatascraper/src/shared/scrapers/USA/AK/tests` 🡒 `USA/AK`
-const scraperNameFromPath = s => s.replace(join(__dirname, '..', 'scrapers'), '').replace('/tests', '');
-
-// Remove geojson + undefined from scraper result
-const strip = d => {
-  delete d.feature;
-  for (const prop of Object.keys(d)) {
-    if (d[prop] === undefined) delete d[prop];
+function scraperNameAndDateFromPath(s) {
+  const parts = s.split(path.sep);
+
+  const dateRegex = /^\d{4}-\d{2}-\d{2}$/;
+  const name = parts.filter(s => !dateRegex.test(s)).join(path.sep);
+  const dt = parts.filter(s => dateRegex.test(s));
+
+  const date = dt.length === 0 ? null : dt[0];
+  return {
+    scraperName: name,
+    date
+  };
+}
+
+/** Changes a URL to a filesystem-friendly name, removes extension. */
+function sanitizeUrl(s) {
+  const ext = path.extname(s);
+  return s
+    .replace(ext, '')
+    .replace(/^https?:\/\//i, '')
+    .replace(/[^a-z0-9]/gi, '_');
+}
+
+/** Run a single scraper test directory. */
+async function runTest(t, cacheRoot, testDirectory) {
+  const { scraperName, date } = scraperNameAndDateFromPath(testDirectory);
+
+  // Monkeypatch global get for this test.
+  // eslint-disable-next-line no-unused-vars
+  get.get = async (url, type, date, options) => {
+    const sanurl = sanitizeUrl(url);
+    const respFile = join(testDirectory, sanurl);
+    // console.log(`  Call: ${url}\n  Sanitized: ${sanurl}\n  Response: ${respFile}`);
+    return fs.readFile(join(cacheRoot, respFile));
+  };
+
+  const pathParts = [__dirname, '..', '..', '..', 'src', 'shared', 'scrapers', scraperName, 'index.js'];
+  const scraperObj = imports(join(...pathParts)).default;
+
+  let result = null;
+  try {
+    process.env.SCRAPE_DATE = date;
+    result = await runScraper.runScraper(scraperObj);
+  } catch (e) {
+    t.fail(`${scraperName} on ${date}, error scraping: ${e}`);
+  } finally {
+    delete process.env.SCRAPE_DATE;
+    get.get = oldGetGet;
```

Comment:
weird naming to have anything like foo.foo

Reply:
Yes, that's the codebase though, I didn't want to refactor it. :-)